pkg/client/restclient/config.go

```diff
@@ -130,10 +130,9 @@ type ContentConfig struct {
 	// a RESTClient directly. When initializing a Client, will be set with the default
 	// code version.
 	GroupVersion *unversioned.GroupVersion
-	// Codec specifies the encoding and decoding behavior for runtime.Objects passed
-	// to a RESTClient or Client. Required when initializing a RESTClient, optional
-	// when initializing a Client.
-	Codec runtime.Codec
+	// NegotiatedSerializer is used for obtaining encoders and decoders for multiple
+	// supported media types.
+	NegotiatedSerializer runtime.NegotiatedSerializer
```

Comment:
If we ever need to do renegotiation (per API) we have to have this.  I'm wondering whether we want to preserve Codec (for admins who don't care).  I guess for simplicity I don't care - @liggitt for opinions